docs/source/cupy-reference/difference.rst

```diff
@@ -0,0 +1,55 @@
+Difference between CuPy and NumPy
+=================================
+
+The interface of CuPy is designed to obey the interface of NumPy.
+However it has some differenece.
+
+
+Cast behavior from float to integer
+-----------------------------------
+
+Some casting behavior from float to integer is not defined.
+For example from a negative float to unsigned integer and infinity to integer.
+It depends on your CPU.
+This is Intel CPU result::
+
+  >>> numpy.array([-1], dtype='f').astype('I')
+  array([4294967295], dtype=uint32)
+  >>> cupy.array([-1], dtype='f').astype('I')
+  array([0], dtype=uint32)
+
+  >>> numpy.array([float('inf')], dtype='f').astype('i')
+  array([-2147483648], dtype=int32)
+  >>> cupy.array([float('inf')], dtype='f').astype('i')
+  array([2147483647], dtype=int32)
+
+Note that NumPy itself behaves diffenently with differenct CPUs.
+
+
+Boolean values squared
+----------------------
+
+In NumPy implementation, ``x ** 2`` is calculated using multiplication operator as ``x * x``.
+Because result of multiplication of boolean values is boolean, ``True ** 2`` return boolean value.
+However, when you use power operator with other arguments, it returns int values.
+In CuPy, a CPU requires to wait GPU in order to check its value, and it is slow.
```

Comment:
If we aligned the behavior of the squared boolean values of CuPy to that of NumPy, we would have to check their types in advance of the calculation. But it would be slow because it would force CPUs to wait until the calculation on GPUs.